File: src/main/java/gargoyle/ct/sound/Audio.java
package gargoyle.ct.sound;

import org.jetbrains.annotations.NotNull;

import java.net.URL;

public final class Audio {
    private Audio() {
    }

    @NotNull
    public static AudioClip play(URL url) {
        AudioClip clip = load(url);
        clip.play();
        return clip;
    }

    public static AudioClip load(URL url) {
        return new AudioClipImpl(url);
    }
}

File: src/main/kotlin/gargoyle/ct/sound/Audio.kt
package gargoyle.ct.sound

import java.net.URL

object Audio {
    fun play(url: URL): AudioClip {
        val clip = load(url)
        clip.play()
        return clip
    }

    @JvmStatic
    fun load(url: URL): AudioClip {
        return AudioClipImpl(url)
    }
}
